apps/drain-safe/src/hooks/useTimeout.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useLayoutEffect, useRef } from 'react';
+
+function useTimeout(callback: () => void, delay: number | null) {
```

Comment:
What is your opinion to avoid useRef here and call directly to the callback if no delay is provided?

```
import { useEffect } from 'react';

function useTimeout(callback: () => void, delay: number | null) {
  useEffect(() => {
    if (!delay && delay !== 0) {
      callback();
      return;
    }

    const id = setTimeout(() => callback(), delay);

    return () => clearTimeout(id);
  }, [delay, callback]);
}

export default useTimeout;

```

Reply:
yep, that's because the outside callback can change because another rerender and with the ref we keep track of the original one